src/features/courses/course-controller.ts

```diff
@@ -2399,16 +2419,18 @@ class CourseController {
             const questions = await CourseWWTopicQuestion.findAll(findOptions);
             if (!_.isNil(courseTopicContentId)){
                 const topic = await this.getTopicById({id: courseTopicContentId});
-                // TODO remove assessment hardcoding
+                // TODO remove assessment hardcoding -- userId nil-check is for TS
```

Comment:
> nil-check is for TS?
What do you mean by this? In other cases it is because of how TS manages scoping, if you nil check and then put it in a closure TS will be wrong and think it is nillable, but in this case it looks like legit concern

Reply:
yeah, I shouldn't have to check it again, but TS was complaining...

Reply:
Yes you should, this is not in the closure for the userId check
if userId is nil the above if block will not run, since it is no longer in that if statement it could once again be nil

This is a valid TS complaint